src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -131,34 +132,75 @@ private function createRelationsForItem($item, $formattedRelations)
+                        if ($field['allow_duplicate_pivots'] ?? false) {
+                            $keyName = $field['pivot_key_name'] ?? 'id';
+                            $sentIds = array_filter(array_column($values, $keyName));
+                            $dbValues = $relation->get()->pluck($keyName)->all();
+                            $toDelete = array_diff($dbValues, $sentIds);
+
+                            if (! empty($toDelete)) {
+                                foreach ($toDelete as $id) {
+                                    $relation->newPivot()->where($keyName, $id)->delete();
+                                }
+                            }
+                            foreach ($values as $value) {
+                                // if it's an existing pivot, update it
+                                if (isset($value[$keyName])) {
+                                    $relation->newPivot()->where($keyName, $value[$keyName])->update($this->preparePivotAttributesForUpdate($value, $relation));
+                                } else {
+                                    $relation->newPivot()->create($this->preparePivotAttributesForCreate($value, $relation, $item->getKey()));
+                                }
+                            }
+                            break;
+                        }
 
-                    $relationValues = [];
+                        $belongsToManyValues = [];
 
-                    if (is_array($values) && is_multidimensional_array($values)) {
                         foreach ($values as $value) {
                             if (isset($value[$relationMethod])) {
-                                $relationValues[$value[$relationMethod]] = Arr::except($value, $relationMethod);
+                                $belongsToManyValues[$value[$relationMethod]] = Arr::except($value, $relationMethod);
                             }
                         }
+
+                        $item->{$relationMethod}()->sync($belongsToManyValues);
+
+                        break;
                     }
 
                     // if there is no relation data, and the values array is single dimensional we have
-                    // an array of keys with no aditional pivot data. sync those.
-                    if (empty($relationValues)) {
-                        $relationValues = array_values($values);
+                    // an array of keys with no additional pivot data. sync those.
+                    if (empty($belongsToManyValues)) {
+                        $belongsToManyValues = array_values($values);
                     }
-
-                    $item->{$relationMethod}()->sync($relationValues);
+                    $item->{$relationMethod}()->sync($belongsToManyValues);
                     break;
             }
         }
     }
 
+    private function preparePivotAttributesForCreate(array $attributes, BelongsToMany $relation, string|int $relatedItemKey)
```

Comment:
I am not sure if we need two distinct functions. Probably we could just have one.

Reply:
Prolly could. What about morphs many